core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -273,7 +274,17 @@ public boolean copyBlock(long blockId, StorageDir dstDir) throws IOException {
       CommonUtils.cleanDirectBuffer(buffer);
     }
     if (copySuccess) {
-      dstDir.addBlockId(blockId, size, mLastBlockAccessTimeMs.get(blockId), true);
+      Long accessTime = mLastBlockAccessTimeMs.get(blockId);
```

Comment:
If you look at the deleteBlock(blockId) method, it removes the access time first and add the blockId to mToRemoveBlockIdSet. These are not atom operations. It is possible the access time is null but mToRemoveBlockIdSet doesn't contains the blockId. Currently the NullPointerException is caused by passing a null access time as a long value to addBlockId method. It seems necessary to check if the access time is null.